### PR TITLE
impl self-closing context handling

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: sql/src/test/java/io/crate/integrationtests/FetchOperationIntegrationTest.java [hunk 1]
```diff
@@ -165,12 +169,20 @@ public void testCollectDocId() throws Exception {
         Planner.Context plannerContext = new Planner.Context(clusterService());
         CollectNode collectNode = createCollectNode(plannerContext, false);
 
+        UUID jobId = UUID.randomUUID();
+        collectNode.jobId(jobId);
```
Comment (seut) [inline sql/src/test/java/io/crate/integrationtests/FetchOperationIntegrationTest.java]: jobId is already set at createCollectNode()

File: sql/src/test/java/io/crate/jobs/JobExecutionContextTest.java [hunk 2]
```diff
@@ -21,78 +21,68 @@
...
+    public void after() throws Exception {
+        threadPool.shutdown();
+        threadPool.awaitTermination(500, TimeUnit.MILLISECONDS);
+    }
 
     @Test
-    public void testInitializingFinalMergeForTheSameExecutionNodeThrowsAnError() throws Exception {
-        expectedException.expect(IllegalStateException.class);
+    public void testAddTheSameContextTwiceThrowsAnError() throws Exception {
+        expectedException.expect(IllegalArgumentException.class);
```
Comment (seut) [inline sql/src/test/java/io/crate/jobs/JobExecutionContextTest.java]: would be great to expect a message here also

**Comment by seut** [comment]
otherwise ack if travis succeed or possible failures are unrelated ;)

**Comment by mfelsche** [comment]
i can only second @seut : "otherwise ack if travis succeed or possible failures are unrelated ;)"

**Pull request merged**